crates/pep508-rs/src/marker/tree.rs

```diff
@@ -1820,60 +1803,75 @@ mod test {
 
     #[test]
     fn test_marker_simplification() {
+        assert_simplifies("python_version == '3.9'", "python_full_version == '3.9.*'");
         assert_simplifies(
-            "python_version == '3.9' or python_version == '3.9'",
-            "python_version == '3.9'",
+            "python_version == '3.9.0'",
+            "python_full_version == '3.9.*'",
+        );
+
+        assert_simplifies("python_version != '3.9'", "python_full_version != '3.9.*'");
+
+        assert_simplifies("python_version >= '3.9.0'", "python_full_version >= '3.9'");
+        assert_simplifies("python_version <= '3.9.0'", "python_full_version < '3.10'");
+
+        assert_simplifies(
+            "python_version == '3.*'",
+            "python_full_version >= '3' and python_full_version < '4'",
```

Comment:
Is there a reason we prefer the later form instead of `python_full_version == '3.*'`?

Reply:
Mostly because I've never seen a bound like this in practice and it didn't seem worth adding a special case for. Can add it if you think it's important.

Reply:
I don't, was just curious since had similar transformations for the `*` patch version case